### ISSUE TITLE
Score column contains square of Euclidean distance?

### ISSUE THREAD

**Issue by tiborsekera** [dotnet/machinelearning#4011]
The column "Score" appears to contain square of Euclidean distance. Is that correct?

E.g., for centroid coordinates [0,0,0] and a prediction coordinate [2,0,0], the "Score" would be 4, and NOT 2.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: d93c6f58-e726-7734-06ed-c3b8f87c6a6e
* Version Independent ID: bcd37e78-d924-c5c2-f5d5-745a8d586330
* Content: [KMeansTrainer Class (Microsoft.ML.Trainers)](https://docs.microsoft.com/en-us/dotnet/api/microsoft.ml.trainers.kmeanstrainer?view=ml-dotnet)
* Content Source: [dotnet/xml/Microsoft.ML.Trainers/KMeansTrainer.xml](https://github.com/dotnet/ml-api-docs/blob/live/dotnet/xml/Microsoft.ML.Trainers/KMeansTrainer.xml)
* Product: **dotnet-ml-api**
* GitHub Login: @sfilipi
* Microsoft Alias: **johalex**

**Comment by najeeb-kazmi** [comment]
@tiborsekera thanks for catching that. The score column is indeed the square of the Euclidean distance. I'll update the docs to reflect that.

**Comment by justinormont** [comment]
Deeper question is, are we better returning the square-of-L2 (as current) or just L2? 

The main use I have for KMeans is in its use as a featurizer. Using the output scores/distances from KMeans as inputs to the final model (generally for classification/regression models). 

![image](https://user-images.githubusercontent.com/4080826/73106707-df50c480-3eb0-11ea-864f-301068a47a33.png)

This would be answered by running benchmarks w/ linear models as the stack's final layer. 

---

I did a quick test on a single dataset (400k rows of the Criteo 1TB dataset). No significant difference was found, with a slight leaning towards keeping the current square-of-L2 instead of L2. 

tldr; current method of returning square-of-L2 seems ok. Nothing to see here.
